### PR TITLE
Modify rule S3504: Adapt to LaYC

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

**Review by victor-diez-sonarsource — CHANGES_REQUESTED** [review]
I think there is space in this rule to explain more concepts. 

When talking about var: https://developer.mozilla.org/en-US/docs/Glossary/Hoisting

Let and const: non-hoisted but you face the [temporal dead zone](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let#temporal_dead_zone_tdz)

Probably explain a little bit about js [scopes](https://developer.mozilla.org/en-US/docs/Glossary/Scope)

**Comment by yassin-kammoun-sonarsource** [comment]
@victor-diez-sonarsource Is it better now ?

File: rules/S3504/javascript/rule.adoc [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+
+
+Variables declared with ``++const++`` are also block-scoped, similar to  ``++let++``. However, ``++const++`` variables are immutable, meaning their value cannot be changed after assignment. This applies to the binding between the variable name and its value, but it does not mean the value itself is immutable if it is an object or an array.
+
+
+A variable declared with ``++let++`` or ``++const++`` is said to be in a "temporal dead zone", meaning the period between entering a scope and declaring a ``++let++`` or ``++const++`` variable. During this phase, accessing the variable results in a ``++ReferenceError++``. This helps catch potential errors and encourages proper variable declaration.
+
+
+Unlike ``++let++`` and ``++const++``, variables declared with ``++var++`` are subject to "hoisting," which means that they are moved to the top of their scope during the compilation phase, even if the actual declaration is placed lower in the code.
```
Comment (victor-diez-sonarsource) [inline rules/S3504/javascript/rule.adoc:14]:
```suggestion
Unlike ``++let++`` and ``++const++``, variables declared with ``++var++`` are subject to "hoisting", which means that they are moved to the top of their scope during the compilation phase, even if the actual declaration is placed lower in the code.
```

**Review by victor-diez-sonarsource — APPROVED** [review]
excellent, thanks! :)

just a small typo with a comma

File: rules/S3504/javascript/rule.adoc (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply rules/S3504/javascript/rule.adoc:14]: 👍

**Pull request merged**